src/main/scala/fs2/aws/kinesis/RecordProcessor.scala

```diff
@@ -0,0 +1,46 @@
+package fs2
+package aws
+package kinesis
+
+import com.amazonaws.services.kinesis.clientlibrary.interfaces._
+import com.amazonaws.services.kinesis.clientlibrary.types._
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.ShutdownReason
+import scala.collection.JavaConverters._
+
+
+private[kinesis] class RecordProcessor(cb: CommittableRecord => Unit) extends v2.IRecordProcessor {
+   private var shardId: String = _
```

Comment:
do we need all those to be vars?

Reply:
this class extends the AWS record processor java abstract class. the `initialize` method receives the input which dictates which shard this record processor is for. do you have an idea on how to capture the data we need and avoid using vars? It would be great to keep it immutable

Reply:
I'll have to pull it down and play with it!